File: 프로그래머스/unrated/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.java
public class Solution {
    public String solution(String[] my_strings, int[][] parts) {
        StringBuilder answer = new StringBuilder();
        for (int i = 0; i < parts.length; i++) {
            answer.append(my_strings[i].substring(parts[i][0], parts[i][1] + 1));
        }
        return answer.toString();
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        String[] my_string = {"progressive", "hamburger", "hammer", "ahocorasick"};
        int[][] parts = {{0, 4}, {1, 2}, {3, 5}, {7, 7}};
        String answer = sol.solution(my_string,parts);
        System.out.println("answer = " + answer);
    }
}

File: JavaScript/프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.js
function solution(my_strings, parts) {
    let answer = '';

    for (let i = 0; i < my_strings.length; i++) {
        let [s, e] = parts[i];
        let substring = my_strings[i].slice(s, e + 1);

        answer += substring;
    }

    return answer;
}